### PR TITLE
Promote CRD field selector e2e test to conformance

### PR DESCRIPTION
**Pull request by jpbetz**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

KEP-4358 is promoting to stable in 1.32

Test history: https://storage.googleapis.com/k8s-triage/index.html?test=CustomResourceFieldSelectors

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/area conformance
@kubernetes/sig-architecture-pr-reviews @kubernetes/sig-api-machinery-pr-reviews @kubernetes/cncf-conformance-wg

### CODE REVIEW

**Comment by aojea** [comment]
@jpbetz the job failure is legit

```
--- test/conformance/testdata/conformance.yaml	2024-10-30 04:27:13.377966386 +0000
+++ _output/conformance.yaml	2024-10-30 04:41:04.149676301 +0000
@@ -332,8 +332,8 @@
   file: test/e2e/apimachinery/custom_resource_definition.go
 - testname: custom-resource-definition-field-selectors-list-watch-register-informers
   codename: '[sig-api-machinery] CustomResourceFieldSelectors [Privileged:ClusterAdmin]
-    [FeatureGate:CustomResourceFieldSelectors] [Beta] CustomResourceFieldSelectors
-    MUST list and watch custom resources matching the field selector [Conformance]'
+    CustomResourceFieldSelectors MUST list and watch custom resources matching the
+    field selector [Conformance]'
   description: Create a Custom Resource Definition with SelectableFields. Create a
     custom resource with two versions and a conversion webhook that converts between
     the versions. Attempt to list, watch and register informers for the custom resources
See instructions in test/conformance/README.md
+++ exit code: 1
+++ error: 1
[0;31mFAILED   verify-conformance-yaml.sh	184s
```
check `test/conformance/README.md` to update the conformance file

**Comment by jpbetz** [comment]
> check test/conformance/README.md to update the conformance file

Huh, I just rebased and still, when I run `hack/update-conformance-yaml.sh` I don't get a diff.  I've manually edited the file to match what CI wants.. but it feels like something is wrong.

File: test/conformance/testdata/conformance.yaml [hunk 1]
```diff
@@ -330,6 +330,21 @@
     resources are available.
   release: v1.16
   file: test/e2e/apimachinery/custom_resource_definition.go
+- testname: custom-resource-definition-field-selectors-list-watch-register-informers
+  codename: '[sig-api-machinery] CustomResourceFieldSelectors [Privileged:ClusterAdmin]
+    [FeatureGate:CustomResourceFieldSelectors] [Beta] CustomResourceFieldSelectors
```
Comment (Jefftree) [inline test/conformance/testdata/conformance.yaml:335]: I guess manually remove L335-336? I'm able to replicate the diff pulling your branch and running `update-conformance-yaml.sh`

Comment (Jefftree) [reply test/conformance/testdata/conformance.yaml:335]: image

**Comment by dims** [comment]
Looks like the few [failures](https://storage.googleapis.com/k8s-triage/index.html?test=CustomResourceFieldSelectors) that show up in triage board are unrelated... ready to land!

/approve

**Pull request merged**